streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -0,0 +1,520 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.internals;
+
+import org.apache.kafka.clients.consumer.CommitFailedException;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.errors.ProducerFencedException;
+import org.apache.kafka.common.utils.Time;
+import org.apache.kafka.streams.errors.StreamsException;
+import org.apache.kafka.streams.processor.TaskId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.util.Collections.singleton;
+
+class TaskManager {
+    // initialize the task list
+    // activeTasks needs to be concurrent as it can be accessed
+    // by QueryableState
+    private static final Logger log = LoggerFactory.getLogger(TaskManager.class);
+    private final Map<TaskId, Task> activeTasks = new ConcurrentHashMap<>();
+    private final Map<TaskId, Task> standbyTasks = new HashMap<>();
+    private final Map<TopicPartition, Task> activeTasksByPartition = new HashMap<>();
+    private final Map<TopicPartition, Task> standbyTasksByPartition = new HashMap<>();
+    private final Set<TaskId> prevActiveTasks = new TreeSet<>();
+    private final Map<TaskId, Task> suspendedTasks = new HashMap<>();
+    private final Map<TaskId, Task> suspendedStandbyTasks = new HashMap<>();
+    private final ChangelogReader changelogReader;
+    private final Time time;
+    private final String logPrefix;
+    private final Consumer<byte[], byte[]> restoreConsumer;
+    private final StreamThread.AbstractTaskCreator taskCreator;
```

Comment:
nit: maybe we can separate AbstractTaskCreator and its two impls into separate classes once we are finalizing the refactoring in a follow-up PR (this PR can stay as is to keep it from exploding LOC)